React/Base/macOS/RCTUIKit.m

```diff
@@ -34,13 +34,14 @@ void UIGraphicsBeginImageContextWithOptions(CGSize size, __unused BOOL opaque, C
 		scale = 1.0;
 	}
 
-	size_t width = ceilf(size.width * scale);
-	size_t height = ceilf(size.height * scale);
+	size_t width = fmax(ceilf(size.width * scale), 1);
+	size_t height = fmax(ceilf(size.height * scale), 1);
 
 	CGColorSpaceRef colorSpace = CGColorSpaceCreateDeviceRGB();
 	CGContextRef ctx = CGBitmapContextCreate(NULL, width, height, 8/*bitsPerComponent*/, width * 4/*bytesPerRow*/, colorSpace, kCGImageAlphaPremultipliedFirst);
```

Comment:
Should we just not do this if the width/height are zero instead of forcing them to be at least 1?

Reply:
Don't do what?   Don't call UIGraphicsBeginImageContextWithOptions at all if width or height is zero?   No: because that would be creating more divergence from the facebook repo.   And in UIKit, it just works.   The only other thing we could do is just *not* RCTAssert in UIGraphicsEndImageContext, but then we loose any protection that UIGraphicsBeginImageContextWithOptions and UIGraphicsEndImageContext are balanced.   But maybe that's ok.

Reply:
Seems like a good divergence that we should contribute back to facebook?

Reply:
I disagree.  We're emulating a set of UIKit functions on macOS.  The UIKit contract allows passing zero width or height without harm.   Refactoring the shared code to avoid this case is unnecessary for iOS and future changes could introduce new cases that don't take into account our emulated methods' shortcomings.   We can either prevent the context from failing to be created as I done, or remove the RCTAssert in UIGraphicsEndImageContext.   I prefer it this way: even though we have a 1x1 bitmap, the RCTGraphicsContextSizeKey ivar is the correct 0x0 and the subsequent calls to UIGraphicsGetImageFromCurrentImageContext will return the appropriate image.

Reply:
My thinking is that diverging to improve code is a good thing even if it means there's a difference (which we'll get rid of whenever we push the fix back to gh/fb).

Reply:
just because iOS currently allows us doesn't mean we should be doing it. Seems like the kind of thing that would "randomly" start crashing in a future version of iOS that tries to get more strict to help us.

Reply:
Sorry I don't think my comments were clear. Can we just not create the context if 0 size on both macOS and iOS? That really seems like something facebook would accept non-controversially?

Reply:
Ok, most of the usages already had non-zero size checks of some kind except a couple places in RCTBorderDrawing.m.   There are a couple other places in test code that don't check, but its non production code.